### PR TITLE
Updating code to be able being built by gcc 6

### PR DESCRIPTION
**Pull request by elbing**
Added -Wno-frame-address to avoid errors about
__builtin_frame_address in:
- libc/port/getcallerpc.c
- libc/port/malloc.c

And related in kernel and cmd.
Unfortunately, clang doesn't recognize this warning supress option,
instead it has -Wno-address. Suggestions for tweaking old build and new
build tools are welcome.

NOTE: if/else statements indentation is now a warning in case
{} would be missing.

### CODE REVIEW

**Comment by elbing** [comment]
Since __builtin_frame_address is used across the code, I suggest including the warning supress option by default in old build tool code and some kind of arrangement in new build tool to check what version of gcc is used.

**Comment by elbing** [comment]
Tested, built and booted in kvm, terminal and cpu server.

**Comment by elbing** [comment]
Seeing travis, clang doesn't pass the check :-(.

File: amd64/include/cflags.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
 			"-Wall",
 			"-Wno-missing-braces",
 			"-Wno-parentheses",
-			"-Wno-unknown-pragmas"
+			"-Wno-unknown-pragmas",
+			"-Wno-frame-address"
```
Comment (rminnich) [inline amd64/include/cflags.json]: If this is gcc-specific it needs to be in toolopts, not here.

**Comment by elbing** [comment]
Removed from sys/src/FLAGS for new build tool, I don't know if we have options like toolopts for mkx.

**Comment by sevki** [comment]
LGTM

**Comment by sevki** [comment]
> NOTE: if/else statements indentation is now a warning in case
{} would be missing. 

I'm liking this a lot. 👍 
missing {} is a pet peeve of mine

**Pull request merged**